control/DomainBlock.js

```diff
@@ -46,10 +46,14 @@ const BlockManager = require('../control/BlockManager.js');
 const blockManager = new BlockManager();
 
 const _ = require('lodash');
+let instance;
 class DomainBlock {
 
   constructor() {
-
+    if (instance == null) {
```

Comment:
this is not necessary, `module.exports = new Class()` will be enough to ensure singleton